packages/trpc/server/routers/viewer/eventTypes.tsx

```diff
@@ -603,4 +611,78 @@ export const eventTypesRouter = router({
         id,
       };
     }),
+  duplicate: eventOwnerProcedure.input(EventTypeDuplicateInput.strict()).mutation(async ({ ctx, input }) => {
+    const { id: originalEventTypeId, title: newEventTitle, slug: newSlug } = input;
+    const eventType = await ctx.prisma.eventType.findUnique({
+      where: {
+        id: originalEventTypeId,
+      },
+      include: {
+        customInputs: true,
+        schedule: true,
+        users: true,
+        team: true,
+        workflows: true,
+        webhooks: true,
+      },
+    });
+
+    if (!eventType) {
+      throw new TRPCError({ code: "NOT_FOUND" });
+    }
+    // Validate user is owner of event type or in the team
```

Comment:
TODO

Reply:
Is this necessary? i am not able to think of a scenario where an user would be able to create an event that is not created by the user itself or it's members. Can you elaborate this?

Reply:
Yeah, that makes a lot of sense, I will try to think about this more carefully in future PRs.
Thanks for explaining @alannnc.

Reply:
No worries, anytime. What you think about this new approach? It's cleaner IMO and if you can tell me if its working for you it will be a lot of help. @harshsinghatz

Reply:
Sorry for late replies, our timezones don't match lol. Your notifications starts popping up when i am half asleep.
Regarding the approach I feel there are more duplications which would have been avoided but i think overall it's more readable.
Have you implemented it completely or is it still in progress?

Reply:
It's complete right now, obviously just missing webhooks/workflows. But we can add that later.